### ISSUE TITLE
Project equality is now hilariously broken when isolated Projects are enabled

### ISSUE THREAD

**Issue by tg-freigmbh** [gradle/gradle#30745]
### Current Behavior

Put this line of code anywhere in any build.gradle.kts:
println(project.rootProject == project.rootProject)
it prints true if isolatedProjects are disabled
it prints false if isolated Projects are enabled

### Expected Behavior

That should usually print true right?

### Self-contained Reproducer Project

println(project.rootProject == project.rootProject)

### Gradle version

8.10.2

### Gradle version that used to work

8.8

**Comment by tg-freigmbh** [comment]
![grafik](https://github.com/user-attachments/assets/a8a87cb1-0a63-4572-bd0e-a72d6aaf1bef)
I guess the equals of ProblemReportingProject should feature an sameObject check as well as an unwrap...

**Comment by ov7a** [comment]
This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

***

Related:
* https://github.com/gradle/gradle/issues/30272

**Comment by mlopatkin** [comment]
This feature request is in the backlog of the relevant team and is prioritized by them.
***

While we cannot promise globally identical project references, we'll fix the `equals` contract.

**Comment by tg-freigmbh** [comment]
Yes == was used in this example because its kotlin script style, it gets converted into equals anyway (at least thats my understanding). Idenntically projects references are not needed on my part, but fixing the euals contract would be very helpful :)

**Comment by hfhbd** [comment]
This breaks the Kotlin MPP plugin when using `isolatedProjects = true`, mostly Kotlin JS that applies a NodeJsRootPlugin and has a check to not apply the RootPlugin to a non root project.
